tcn_hpl/data/components/augmentations.py

```diff
@@ -181,108 +291,238 @@ def init_delta(self):
     def forward(self, features):
         delta = self.init_delta()
 
-        if self.feat_version == 1:
-            activation_idxs = range(features.shape[1])
-
-        elif self.feat_version == 2:
-            num_obj_feats = self.num_obj_classes - 2  # not including hands in count
-            num_obj_points = num_obj_feats * 2
-
-            obj_acts_idx = num_obj_points + 1 + num_obj_points + 2 + 1
-            activation_idxs = [0, num_obj_points + 1] + list(
-                range(obj_acts_idx, features.shape[1])
-            )
-
-        elif self.feat_version == 3:
-            activation_idxs = [0, 3] + list(range(7, features.shape[1], 5))
-
-        elif self.feat_version == 5:
-            num_obj_feats = self.num_obj_classes - 2  # not including hands in count
-            num_obj_points = num_obj_feats * 2
-
-            obj_acts_idx = num_obj_points + 1 + num_obj_points + 2 + 1 + 1
-            activation_idxs = [0, num_obj_points + 1] + list(
-                range(obj_acts_idx, features.shape[1], 3)
-            )
-
-        else:
-            NotImplementedError(f"Unhandled version '{self.feat_version}'")
+        for i in range(features.shape[0]):
+            frame = features[i]
```

Comment:
I sort of wish "frame" and "features" switched names, or "frame" was instead called "single_frame_features" or "frame_features"